tests/integration_tests/utils.py

```diff
@@ -758,3 +759,273 @@ def filter(stats):
     finally:
         # Remove results/intermediate data saved to disk
         shutil.rmtree(output_dir, ignore_errors=True)
+
+
+class ParameterUpdateError(Exception):
+    pass
+
+
+# TODO: do we need this version of parameter update checking
```

Comment:
Could this method be consolidated with `_assert_module_parameters_updated`? The two methods look similar.

Reply:
removed redundant code